docs/docs/reference/frontend-styleguide.md

```diff
@@ -0,0 +1,423 @@
+---
+myst:
+  html_meta:
+    "description": "Frontend Styleguide"
+    "property=og:description": "Frontend Styleguide"
+    "property=og:title": "Frontend Styleguide"
+    "keywords": "Plone, reference, styleguide, frontend"
+---
+
+# Frontend Styleguide
+
+This part of the documentation contains reference material for the frontend styleguide, including design principles, UI components, and best practices.
+It is intended to be followed in all kitconcept projects and add-ons.
+
+## Use TypeScript
+
+We use TypeScript for all new frontend code.
+Typings does not have to be perfect and crystal correct, but this alone, should be better than using plain JavaScript.
+This helps to catch errors early and improves code quality.
```

Comment:
Is creating files with .ts / .tsx extension enough, or do we have to do something more? Do we have to do something to enable type checking in our editor?